### PR TITLE
[opt](function) change array_zip function null literal to array<null>

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/FunctionCallExpr.java [hunk 1]
```diff
@@ -1895,8 +1895,9 @@ && collectChildReturnTypes()[0].isDecimalV3()) {
                 // array_zip varargs special case array_zip(array1, array2, ...)
                 // we only specialize array_zip with first array type, next type we same with custom type
                 if (i >= args.length && (fnName.getFunction().equalsIgnoreCase("array_zip"))) {
-                    if (argTypes[i].isNull()) {
-                        uncheckedCastChild(args[i - 1], i);
+                    if (argTypes[i].isNull()) { //select array_zip(['a', 'b', 'c'], NULL, [11, 22, 33]);
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/FunctionCallExpr.java:1898]: Do not use the way to solve problem

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Pull request closed without merging**